### PR TITLE
1450 goback stoprun

### PR DESCRIPTION
**Pull request by LANOYDO**

### CODE REVIEW

File: TypeCobol.Test/Parser/CodeElements/STOP.txt [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+Line 15[1,4] <37, Warning, General> - Warning: GOBACK should be used instead of STOP RUN RuleStack=codeElement>stopStatement,  OffendingSymbol=[1,4:STOP]<STOP>
+Line 16[1,4] <37, Warning, General> - Warning: GOBACK should be used instead of STOP RUN RuleStack=codeElement>stopStatement,  OffendingSymbol=[1,4:STOP]<STOP>
+Line 17[1,4] <37, Warning, General> - Warning: GOBACK should be used instead of STOP RUN RuleStack=codeElement>stopStatement,  OffendingSymbol=[1,4:STOP]<STOP>
+Line 18[1,4] <37, Warning, General> - Warning: GOBACK should be used instead of STOP RUN RuleStack=codeElement>stopStatement,  OffendingSymbol=[1,4:STOP]<STOP>
+Line 19[1,4] <37, Warning, General> - Warning: GOBACK should be used instead of STOP RUN RuleStack=codeElement>stopStatement,  OffendingSymbol=[1,4:STOP]<STOP>
+Line 20[1,4] <37, Warning, General> - Warning: GOBACK should be used instead of STOP RUN RuleStack=codeElement>stopStatement,  OffendingSymbol=[1,4:STOP]<STOP>
+Line 22[1,4] <37, Warning, General> - Warning: GOBACK should be used instead of STOP RUN RuleStack=codeElement>stopStatement,  OffendingSymbol=[1,4:STOP]<STOP>
```
Comment (maxime645) [inline TypeCobol.Test/Parser/CodeElements/STOP.txt]: The warning should be displayed only on keywords "STOP RUN" not on any "STOP (something)"

Comment (LANOYDO) [reply TypeCobol.Test/Parser/CodeElements/STOP.txt]: fixed 23/09/2019
in class StopStatementChecker with the method OnCodeElement

**Comment by maxime645** [comment]
Rebased version of #1474

File: TypeCobol/Compiler/Diagnostics/CodeElementCheckers.cs [hunk 2]
```diff
@@ -289,6 +289,15 @@ public static void OnCodeElement(StartStatement statement, CodeElementsParser.St
         }
     }
 
+    class StopStatementChecker
+    {
+        public static void OnCodeElement(StopStatement statement, CodeElementsParser.StopStatementContext context)
+        {
+            if (statement.StopRun != null && statement.StopRun.Value)
+                DiagnosticUtils.AddError(statement, "GOBACK should be used instead of STOP RUN", context, MessageCode.Warning);
```
Comment (smedilol) [inline TypeCobol/Compiler/Diagnostics/CodeElementCheckers.cs]: There is no need to display the RuleStack for this error.
So the parameter "context" can be removed.

Comment (LANOYDO) [reply TypeCobol/Compiler/Diagnostics/CodeElementCheckers.cs]: done

Comment (LANOYDO) [reply TypeCobol/Compiler/Diagnostics/CodeElementCheckers.cs]: done
https://github.com/TypeCobolTeam/TypeCobol/tree/08ca1ff0a45ef945f24534bde9a3d5535cf5d2ab

File: TypeCobol.Test/Parser/CodeElements/STOP.txt [hunk 3]
```diff
@@ -1,4 +1,5 @@
 --- Diagnostics ---
+Line 1[1,4] <37, Warning, General> - Warning: GOBACK should be used instead of STOP RUN OffendingSymbol=[1,4:STOP]<STOP>
```
[the remark this answers is not in the document]
Comment (smedilol) [reply TypeCobol.Test/Parser/CodeElements/STOP.txt]: My mistake it's not the Antlr context but the token here.

**Pull request merged**